projects/conda-forge-simulation-stack.yml

```diff
@@ -0,0 +1,54 @@
+---
+name: Packaging the HEP simulation stack on conda-forge
+postdate: 2023-09-05
+categories:
+  - Analysis tools
+  - Open science
+durations:
+  - 3 months
+skillset:
+  - Python
+  - C++
+  - Docker
+status:
+  - Available
+project:
+  - IRIS-HEP
+experiments:
+  - ATLAS
+  - LHCb
+  - CMS
+  - ALICE
+  - EIC
+  - HLLHC
+program:
+  - IRIS-HEP fellow
+location:
+  - Remote
+commitment:
+  - Full time
+shortdescription: Package the HEP simulation tools for conda-forge
+description: >
+  One common toolchain used in high energy physics for simulation is:
+  [MadGraph5_aMC@NLO](https://launchpad.net/mg5amcnlo) to
+  [PYTHIA8](https://pythia.org/) to [Delphes](https://github.com/delphes/delphes).
+  Installing these tools can be challenging at times, especially for new users.
+  [Conda-forge](https://conda-forge.org/) allows for distribution of arbitrarily
+  complex binaries across multiple platforms though the conda/mamba/micromamba/pixi
+  package management ecosystem.
+  As [ROOT](https://anaconda.org/conda-forge/root/) has been successfully packaged and
+  distributed on conda-forge along with the
+  [PYTHIA8 Python bindings](https://anaconda.org/conda-forge/pythia8/) it should be
+  possible to package all the components of the HEP simulation stack and distribute
+  them on conda-forge.
+
+  However, the interconnected nature of some of these tools requires that multiple
+  dependencies are first packaged and distributed before the full stack can be.
+  This project would attempt to package as many of the dependencies of the HEP
+  simulation stack on conda-forge as possible starting with
+  [FastJet](http://fastjet.fr/), [LHAPDF](https://lhapdf.hepforge.org/),
+  [HepMC2](https://gitlab.cern.ch/hepmc/HepMC), and
+  [HepMC3](https://gitlab.cern.ch/hepmc/HepMC3).
+contacts:
+  - name: Matthew Feickert
+    email: matthew.feickert@cern.ch
```

Comment:
@henryiii, @chrisburr, @agbuckley would you be willing to be listed as mentors on this project, if "mentors" here translates to "people that can be emailed and asked to give advice on why things are breaking horribly in the build and then approve and merge PRs/MRs to the projects they maintain"?

Reply:
Hi @matthewfeickert , realistically I am just too overloaded to add this to the list at the moment -- after coming back from one day offline at another workshop to find 200+ departmental emails and our MSc programmes metaphorically on fire. I don't mind being copied in and helping as and when, but it can't be a priority for me and I wouldn't want to advertise as being available and then disappoint. Is there a particular project you had in mind for me to accept MRs on? Obviously the Rivet etc. teams manage our own trackers, and we have a responsibility to help... though Conda has correlated strongly with borked environments run by users who can't provide a MWE.

Chris Gutschow is also pretty technically invested in the simulation stack -- not for conda, but _might_ be available to help on generic things as well as on the specific projects. Not sure if he has a GitHub account.

Reply:
Unfortunately I'm in a similar situation to Andy so can't really commit to reliably responding to questions from the mentee. Though if you let me know when the project starts I can try to keep an eye on `conda-forge/staged-recipes` to help get PRs accepted.

Reply:
> I don't mind being copied in and helping as and when, but it can't be a priority for me and I wouldn't want to advertise as being available and then disappoint

Totally understand this @agbuckley. :+1:

> Is there a particular project you had in mind for me to accept MRs on?

I had originally thought for HepMC, but given @chrisburr's comment https://github.com/research-software-collaborations/project_database/pull/62#discussion_r1316549515 it seems that these are already taken care of! So I think realistically just LHAPDF, and only if that required some patch to get applied to be able to build in conda-forge's conda-smithy CI.

> Obviously the Rivet etc. teams manage our own trackers, and we have a responsibility to help...

The goal of this project was to basically have teams like Rivet _not_ have to do any work, beyond maybe reviewing a MR and saying "LGTM" (as well as approving that their code be distributed on conda-forge).

> though Conda has correlated strongly with borked environments run by users who can't provide a MWE.

Yeah, hopefully this project would have a RSE-inclined Fellow be able to help make conda environments less difficult for the simulation stack to work with.

> Chris Gutschow is also pretty technically invested in the simulation stack

Ah great! I'll follow up with him. Thanks!

---

> Though if you let me know when the project starts I can try to keep an eye on conda-forge/staged-recipes to help get PRs accepted.

@chrisburr, this is more than enough help. Many thanks!

Reply:
Hiya - getting Rivet 3.2 + YODA2 release ready currently has higher priority on my to-do list, but I could probably spend some time on this afterwards. ✌️

Reply:
Sounds good @20DM. To be clear, this is a proposed IRIS-HEP Fellow project for the future, and wouldn't start until either late 2023 or early 2024.

There's already some packages on conda-forge, which is great, but this project would hopefully get things like https://github.com/conda-forge/staged-recipes/pull/21052 finished (if we can't get it done before then) and then iteratively address others so that we could ideally start having the larger applications that pull in many dependencies get packaged as well.